__tests__/container.test.js

```diff
@@ -0,0 +1,192 @@
+
+    const createdContainer = await containerService.createContainer('createByParams', 'someHouseholdId')
+    expect(createdContainer.id).toBeDefined()
+    expect(createdContainer.toDocument()).toMatchObject({
+        name: 'createByParams',
+        householdId: 'someHouseholdId',
+        foodItems: []
+    })
+})
+
+//GetContainer
+test('should get a container that has been created already without errors', async () => {
+    const services = setup()
+    const [containerService] = services
+
+    const containerCreated = await containerService.createContainer('getContainerTest1', 'someHouseholdId')
+    const retrievedContainer = await containerService.getContainerById(containerCreated.id);
+
+    expect(retrievedContainer).toMatchObject(containerCreated)
+    
+})
+
+test('should try to get a container that doesn\'t exist and return null without error', async () => {
+    const services = setup()
+    const [containerService] = services
+
+    const containerId = 'I_Do_Not_Exist'
+    const retrievedContainer = await containerService.getContainerById(containerId);
+
+    expect(retrievedContainer).toBe(null)
+
+})
+
+//DeleteContainer
+test('should create and delete a container by id without error', async () => {
+    const services = setup()
+    const [containerService] = services
+
+    const newContainer = await containerService.createContainer('toBeDeleted', 'toBeDeleted')
+    const containerCreated = await containerService.getContainerById(newContainer.id)
+    expect(containerCreated).toMatchObject(newContainer)
+
+    const deletedContainer = await containerService.deleteContainerById(newContainer.id)
+    expect(deletedContainer).toMatchObject(newContainer)
+
+    const containerAfterDeletion = await containerService.getContainerById(newContainer.id)
+    expect(containerAfterDeletion).toBe(null)
+})
+
+test('should try to delete container by id that doesn\'t exist', async () => {
+    const services = setup()
+    const [containerService] = services
+
+    const containerId = 'I_Do_Not_Exist'
+    const deletedContainer = await containerService.deleteContainerById(containerId)
+    expect(deletedContainer).toBe(null)
+
+})
+
+test('should create and delete a container by object without error', async () => {
+    const services = setup()
+    const [containerService] = services
+
+    const newContainer = await containerService.createContainer('toBeDeleted', 'toBeDeleted')
+    const containerCreated = await containerService.getContainerById(newContainer.id)
+    expect(containerCreated).toMatchObject(newContainer)
+
+    const deletedContainer = await containerService.deleteContainerByObject(newContainer)
+    expect(deletedContainer).toMatchObject(newContainer)
+
+    const containerAfterDeletion = await containerService.getContainerById(newContainer.id)
+    expect(containerAfterDeletion).toBe(null)
+
+})
+
+test('should try to delete a container by object that has a null id', async () => {
```

Comment:
So do you imagine them actually using this functionality? Is this like a `FindAndDelete` call? Where we would be matching the partial object received as a parameter?

Reply:
so this test is just a safety precaution. I anticipate that the frontend uses the containerService.createContiainer(name, householdId) to create their containers but in the cases they just make a Container via its constructor (which by default gives it a null id), this ensures the app doesn't break.

Reply:
Also I would like to discuss throwing errors as part of our backend code